fast64_internal/f3d/f3d_writer.py

```diff
@@ -2024,6 +2027,49 @@ def normToSigned8Vector(normal):
 	return [int.from_bytes(int(value * 127).to_bytes(1, 'big',
 		signed = True), 'big') for value in normal]
 
+def saveBitGeoF3DEX2(value, defaultValue, flagName, geo, matWriteMethod):
+	if value != defaultValue or matWriteMethod == GfxMatWriteMethod.WriteAll:
+		if value:
+			geo.setFlagList.append(flagName)
+		else:
+			geo.clearFlagList.append(flagName)
+
+def saveGeoModeDefinitionF3DEX2(fMaterial, settings, defaults, matWriteMethod):
+	geo = SPGeometryMode([],[])
+
+	saveBitGeoF3DEX2(settings.g_zbuffer, defaults.g_zbuffer, 'G_ZBUFFER',
+		geo, matWriteMethod)
+	saveBitGeoF3DEX2(settings.g_shade, defaults.g_shade, 'G_SHADE',
+		geo, matWriteMethod)
+	saveBitGeoF3DEX2(settings.g_cull_front, defaults.g_cull_front, 'G_CULL_FRONT',
+		geo, matWriteMethod)
+	saveBitGeoF3DEX2(settings.g_cull_back,  defaults.g_cull_back, 'G_CULL_BACK',
+		geo, matWriteMethod)
+	saveBitGeoF3DEX2(settings.g_fog, defaults.g_fog, 'G_FOG', geo, matWriteMethod)
+	saveBitGeoF3DEX2(settings.g_lighting, defaults.g_lighting, 'G_LIGHTING',
+		geo, matWriteMethod)
+
+	# make sure normals are saved correctly.
+	saveBitGeoF3DEX2(settings.g_tex_gen, defaults.g_tex_gen, 'G_TEXTURE_GEN',
+		geo, matWriteMethod)
+	saveBitGeoF3DEX2(settings.g_tex_gen_linear, defaults.g_tex_gen_linear,
+		'G_TEXTURE_GEN_LINEAR', geo, matWriteMethod)
+	saveBitGeoF3DEX2(settings.g_shade_smooth, defaults.g_shade_smooth,
+		'G_SHADING_SMOOTH', geo, matWriteMethod)
+	saveBitGeoF3DEX2(settings.g_clipping, defaults.g_clipping, 'G_CLIPPING',
+		geo, matWriteMethod)
+
+	if len(geo.clearFlagList) == 0:
+		geo.clearFlagList.append('0')
+	elif len(geo.setFlagList) == 0:
+		geo.setFlagList.append('0')
+
+	if matWriteMethod == GfxMatWriteMethod.WriteAll:
+		fMaterial.material.commands.append(SPLoadGeometryMode(geo.setFlagList))
+	else:
+		fMaterial.material.commands.append(geo)
+		fMaterial.revert.commands.append(SPGeometryMode(geo.setFlagList, geo.clearFlagList))
```

Comment:
yeah that's more understandable, I missed the revert before
still I'm pretty sure this can just be:
```suggestion
	fMaterial.material.commands.append(geo)
	if matWriteMethod == GfxMatWriteMethod.WriteDifferingAndRevert:
		fMaterial.revert.commands.append(SPGeometryMode(geo.setFlagList, geo.clearFlagList))
```

also shouldn't the last line be
```py
		fMaterial.revert.commands.append(SPGeometryMode(geo.clearFlagList, geo.setFlagList))
```
(notice the swap of clear/set)

Reply:
the swap is intentional, since its supposed to be the inverse

Reply:
also while what you are saying is correct, I want to use LoadGeometryMode when possible

Reply:
I guess someone else can weigh in, personally I think it's fine. Thoughts? @thecozies